FlowCal/transform.py

```diff
@@ -129,53 +148,98 @@ def to_rfi(data, channels=None, max_range=None, amplification_type=None):
+        elif hasattr(max_range, '__iter__'):
+            # If it's a list, it should be the same length as channels
+            if len(max_range) != len(channels):
+                raise ValueError("channels and max_range should have "
+                    "the same length")
+        else:
+            # If it's not a list or None, raise error
+            raise ValueError("channels and max_range should have the "
+                    "same length")
 
     # Convert channels to integers
     if hasattr(data, '_name_to_index'):
         channels = data._name_to_index(channels)
     else:
         channels = channels
 
-    # If max_range is not specified, take it from data.domain
-    if max_range is None:
-        if hasattr(data, 'domain'):
-            max_range = [len(data.domain(channel)) for channel in channels]
-        else:
-            raise ValueError('max_range should be specified')
-    
-    # If amplification_type is not specified, take it from
-    # data.amplification_type
-    if amplification_type is None:
-        if hasattr(data, 'amplification_type'):
-            amplification_type = data.amplification_type(channels)
-        else:
-            raise ValueError('amplification_type should be specified')
-
-    # Check that channels, max_range, and amplification_type have the same
-    # length
-    if len(channels) != len(max_range):
-        raise ValueError("channels and max_range should have the same length")
-    if len(channels) != len(amplification_type):
-        raise ValueError("channels and amplification_type should have the same"
-            " length")
-
     # Copy data array
     data_t = data.copy().astype(np.float64)
 
     # Iterate over channels
-    for channel, r, at in zip(channels, max_range, amplification_type):
-        # Don't do anything if data is in linear scale
+    for channel, r, at, ag in \
+            zip(channels, max_range, amplification_type, amplifier_gain):
+        # If amplification type is None, try to obtain from data
+        if at is None:
+            if hasattr(data, 'amplification_type'):
+                at = data.amplification_type(channel)
+            else:
+                raise ValueError('amplification_type should be specified')
+        # Define transformation, depending on at[0]
         if at[0]==0:
-            continue
-        # Define transformation
-        tf = lambda x: at[1] * 10**(at[0]/float(r) * x)
+            # Linear amplifier
+            # If no amplifier gain has been specified, try to obtain from data,
+            # otherwise assume one
+            if ag is None:
+                if hasattr(data, 'amplifier_gain'):
+                    ag = data.amplifier_gain(channel)
+                    # If the linear gain has not been specified, it should be
+                    # assumed to be one.
+                    if ag is None:
+                        ag = 1.
+                else:
+                    ag = 1.
```

Comment:
this whole `if-else` segment seems like it could possibly be more elegantly done with a `try-except`; just try calling `data.amplifier_gain`, and `except: ag=1`. this is minor, though